### PR TITLE
Add tests for Hasura query collections/endpoints initialization

### PR DESCRIPTION
**Pull request by ypc-faros**

### CODE REVIEW

File: .tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (cjwooo) [inline .tsconfig.json:1]: Can we merge this with https://github.com/faros-ai/faros-community-edition/blob/main/init/.tsconfig.json

File: init/test/tsconfig.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
-  "extends": "../.tsconfig.json",
+  "extends": "../../.tsconfig.json",
```
Comment (tovbinm) [inline init/test/tsconfig.json:2]: do we have this problem in all the projects? can you please fix?

File: init/test/hasura/init.test.ts [hunk 3]
```diff
@@ -0,0 +1,212 @@
+import axios from 'axios';
+import {mocked} from 'jest-mock';
+import path from 'path';
+import pino from 'pino';
+
+import {HasuraInit} from '../../src/hasura/init';
+import {Endpoint, QueryCollection} from '../../src/hasura/types';
+
+jest.mock('axios');
```
Comment (tovbinm) [inline init/test/hasura/init.test.ts:9]: I personally like using `nock` to mock the underlying service. That way the client actually makes the http call which is intercepted. Example - https://github.com/faros-ai/faros-community-edition/blob/main/init/test/airbyte/index.test.ts

File: init/test/hasura/init.test.ts [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+
+describe('init', () => {
+  const mockedAxios = mocked(axios, true);
+
+  const logger = pino({
+    name: 'hasura-init',
+  });
+
+  const hasura = new HasuraInit(
```
Comment (tovbinm) [inline init/test/hasura/init.test.ts:18]: so many helper methods methods make it difficult to read and understand the tests. lets move the helper methods out to a separate file

File: .tsconfig.json (hunk 1 above)
Comment (tovbinm) [reply .tsconfig.json:1]: yes please, let's have only one style.

**Review by ypc-faros — COMMENTED** [review]
a.gql is actually needed. resources/hasura/endpoints/*.gql are the files containing the queries we want to expose as rest endpoints in Hasura

I should have probably said that somewhere :)

**Comment by tovbinm** [comment]
Oops, my bad. I should not have touched it. --

-Matthew

File: init/test/hasura/init.test.ts (hunk 3 above)
Comment (ypc-faros) [reply init/test/hasura/init.test.ts]: Thanks! `nock` is great 👍  !

File: init/test/hasura/init.test.ts (hunk 4 above)
Comment (ypc-faros) [reply init/test/hasura/init.test.ts:25]: Done, thanks!

**Review by tovbinm — APPROVED** [review]

File: init/src/hasura/init.ts [hunk 5]
```diff
@@ -491,7 +492,7 @@ async function main(): Promise<void> {
       baseURL: args[0],
       headers: {'X-Hasura-Role': 'admin'},
     }),
-    logger
+    pino(logger)
```
Comment (cjwooo) [inline init/src/hasura/init.ts:495]: Why the added pino call here?

Comment (ypc-faros) [reply init/src/hasura/init.ts]: Oops, removed 👍

File: init/test/tsconfig.json (hunk 2 above)
Comment (tovbinm) [reply init/test/tsconfig.json:2]: ok, thanks

**Pull request merged**